utilities/game.py

```diff
@@ -19,3 +20,25 @@ def is_red() -> bool:
 
 def get_team() -> wpilib.DriverStation.Alliance:
     return wpilib.DriverStation.getAlliance()
+
+
+def pose3d_to_translation2d(pose3d: Pose3d) -> Translation2d:
+    return Translation2d(pose3d.x, pose3d.y)
+
+
+def find_closest_tag(robot_pose: Pose2d) -> tuple[Pose3d, int]:
+    tag_ids = [1, 2, 3] if is_red() else [6, 7, 8]
+
+    # Use the first tag to set a baseline for distance
+    best_id = tag_ids[0]
+    best_tag = apriltag_layout.getTagPose(best_id)
+    closest_distance = robot_pose.translation().distance(pose3d_to_translation2d(best_tag))  # type: ignore
+
+    for tag_id in tag_ids[1:]:
+        tag = apriltag_layout.getTagPose(tag_id)
+        d = robot_pose.translation().distance(pose3d_to_translation2d(tag))  # type: ignore
+        if d < closest_distance:
+            closest_distance = d
+            best_tag = tag
+            best_id = tag_id
+    return best_tag, best_id  # type: ignore
```

Comment:
I think it might be better to assert that the best tag is none, so the type checker works. This part of the function is static. It should never break without someone changing it, and its something I think we want to know about instead of ignoring it.

I also thought I would highlight that there is an existing api to do what you are trying to do with the transform to 2d but if you just prefer what you have then that's cool too.

TLDR - I think we should assert that the tag is not none because its a static thing and should never break under reasonable conditions but idm if you prefer to keep your existing to 2d function. I will approve after this is done or I am happy to change it myself

```suggestion
def find_closest_tag(robot_pose: Pose2d) -> tuple[Pose3d, int]:
    tag_ids = [1, 2, 3] if is_red() else [6, 7, 8]

    # Use the first tag to set a baseline for distance
    best_id = tag_ids[0]
    best_tag = apriltag_layout.getTagPose(best_id)
    
    assertbest_tag != None
    
    closest_distance = robot_pose.translation().distance(
        best_tag.toPose2d().translation()
    )

    for tag_id in tag_ids[1:]:
        tag = apriltag_layout.getTagPose(tag_id)
        d = robot_pose.translation().distance(
        tag.toPose2d().translation()
    )
        if d < closest_distance:
            closest_distance = d
            best_tag = tag
            best_id = tag_id
    return best_tag, best_id
```